RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc

```diff
@@ -564,7 +563,8 @@ void MuonIdProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
 
         for (auto& muon : *outputMuons) {
           if (muon.innerTrack().get() == trackerMuon.innerTrack().get() &&
-              cos(phiOfMuonIneteractionRegion(muon) - phiOfMuonIneteractionRegion(trackerMuon)) > 0) {
+              std::abs(reco::deltaPhi(phiOfMuonInteractionRegion(muon), phiOfMuonInteractionRegion(trackerMuon))) <
+                  0.5 * M_PI) {
```

Comment:
make it float and use M_PI_2
so float(M_PI_2): the compiler should optimize....